src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+            this.inputJavaFileTest = RESOURCE_DIR + TEST_DIR + classDTO.getSmallPath()
+                    .split("/")[1] + "/" + classDTO.getSmallName() + ".java";
+            this.inputJavaFileManual = RESOURCE_DIR + MANUAL_DIR + classDTO.getSmallPath()
+                    + classDTO.getSmallName() + ".java";
+            this.generatedClassFullName = TEST_CLASS_PACKAGE_NAME + classDTO.getSmallPath()
+                    .split("/")[1] + "." + classDTO.getSmallName();
+            this.manualClassFullName = MANUAL_CLASS_PACKAGE_NAME + classDTO.getSmallPath().
+                    replace("/", ".") + classDTO.getSmallName();
+            this.jsonSourcePath = classDTO.getJsonPath();
+            this.url = classDTO.getTestURL();
+            this.checkLocatorUniqueness = classDTO.isCheckLocatorUniqueness();
+            this.forceGenerateFiles = classDTO.isForceGenerateFiles();
+            runTestCase();
+        }
+    }
+
+    private void runTestCase() throws Exception {
+        compileTestClasses();
+        testNameOfClass();
+        testAnnotationOfClass();
+        testModifiersOfClass();
+        testSuperClassOfClass();
+        testFields();
+    }
+
+    private void testNameOfClass() throws Exception {
+        String[] testClassName = testClass.getName().split("\\.");
+        String[] manualClassName = manualClass.getName().split("\\.");
+        assertEquals("Different class names", testClassName[testClassName.length - 1],
+                manualClassName[manualClassName.length - 1]);
+    }
+
+    private void testAnnotationOfClass() {
+        List<Annotation> testClassAnnotations = Arrays.asList(testClass.getDeclaredAnnotations());
+        List<Annotation> manualClassAnnotations = Arrays.asList(manualClass.getDeclaredAnnotations());
+        assertEquals("Different class annotations", manualClassAnnotations,
+                testClassAnnotations);
+    }
+
+    private void testModifiersOfClass() {
+        int testMods = testClass.getModifiers();
+        int manualMods = manualClass.getModifiers();
+        assertEquals("Different class modifiers", testMods, manualMods);
+    }
+
+    private void testSuperClassOfClass() {
+        Class<?> testSuperClass = testClass.getSuperclass();
+        Class<?> manualSuperClass = manualClass.getSuperclass();
+        assertEquals("Different superclasses", testSuperClass, manualSuperClass);
+    }
+
+    private void testFields() {
+        List<Field> testFields = Arrays.asList(testClass.getDeclaredFields());
+        List<Field> manualFields = Arrays.asList(manualClass.getDeclaredFields());
+        assertEquals("Different number of fields", testFields.size(),
+                manualFields.size());
+
+        Iterator<Field> testIt = testFields.iterator();
+        Iterator<Field> manualIt = manualFields.iterator();
+
+        while (testIt.hasNext()) {
+            Field testCurrentField = testIt.next();
+            Field manualCurrentField = manualIt.next();
```

Comment:
I wonder if class generator may generate class with unpredictable order of fields. If it may, then test case should be ready for that behaviour. Add javadoc comment to clarify this case.